### ISSUE TITLE
Unable to set null BelongsTo

### ISSUE THREAD

**Issue by ArneSaknussemm89** [flutterdata/flutter_data#113]
What would be the ideal way to set a BelongsTo value to null? If I try any of these ways it doesn't seem like the change persists and the value stays what it was before.

Saving code:
```dart
await context
  .read(userPreferencesRepositoryProvider)
  .save(
    updated.was(preferences),
    remote: false,
  );
```

My first thought to try:
```dart
final updated = branch != null 
  ? preferences.copyWith(
       branch: BelongsTo(branches.where((element) => element.locationName == branch).first),
     )
  : preferences.copyWithNull(branch: true);
```

I've also tried:
```dart
final updated = preferences
  .copyWith(
    branch: branch != null 
      ? BelongsTo<Branch>(branches.where((element) => element.locationName == branch).first) 
      : BelongsTo<Branch>(null)
);
```

But whether I set the value to `null` or `BelongsTo<Branch>(null)` after the save the returned object still has the old BelongsTo value. If I change it to any other value that's not `null` it saves just fine.

**Comment by ArneSaknussemm89** [comment]
Just wanted to bump this!

**Comment by frank06** [comment]
@ArneSaknussemm89 sorry for taking so long! 

Just checked `belongs_to_test.dart` and the way we do this is by mutating the `value` of the relationship.

```dart
preferences.branch.value = null;
```

Does that work?

I will keep this open to have a look at why copying an empty `BelongsTo` is not working in your case.

**Comment by ArneSaknussemm89** [comment]
@frank06 That works!

I'll close this for now, but if you come across why the copyWith method isn't quite working I'd be very interested to know how to do it in an immutable way.

**Comment by frank06** [comment]
@ArneSaknussemm89 pushed a new API so you can do

```dart
final updated = preferences
  .copyWith(
    branch: branch != null 
      ? BelongsTo<Branch>(branches.where((element) => element.locationName == branch).first) 
      : BelongsTo.remove()
);
```

We can't use `null` as the default initializer `BelongsTo()` would be equivalent to `BelongsTo(null)`

**Comment by frank06** [comment]
@ArneSaknussemm89 In case you're interested FD 1.0 has been released (with support for Riverpod 1.0 and more) and the new documentation site is live: https://flutterdata.dev/
